Index: engine/Shopware/Controllers/Backend/Emotion.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- engine/Shopware/Controllers/Backend/Emotion.php	(date 1540825291000)
+++ engine/Shopware/Controllers/Backend/Emotion.php	(date 1541163784000)
@@ -262,7 +262,7 @@
         $emotionId = $this->Request()->get('emotionId');

         if (!$emotionId) {
-            echo 'Parameter emotionId not found!';
+            $this->Response()->setBody('Parameter emotionId not found!');

             return;
         }
@@ -274,7 +274,7 @@
             /** @var string $exportFilePath */
             $exportFilePath = $exporter->export($emotionId);
         } catch (\Exception $e) {
-            echo $e->getMessage();
+            $this->Response()->setBody($e->getMessage());

             return;
         }
@@ -286,10 +286,12 @@
             ->setHeader('Content-disposition', 'attachment; filename="' . basename($exportFilePath) . '"')
             ->sendHeaders();

+        if (!$this->Front()->getParam('disableOutputBuffering')) {
+            ob_end_clean();
+        }
+
         readfile($exportFilePath);
         $this->container->get('file_system')->remove($exportFilePath);
-
-        exit;
     }

     /**
Index: engine/Shopware/Controllers/Backend/MediaManager.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- engine/Shopware/Controllers/Backend/MediaManager.php	(date 1540825291000)
+++ engine/Shopware/Controllers/Backend/MediaManager.php	(date 1541163784000)
@@ -64,7 +64,7 @@
      *
      * @var \Shopware\Components\Model\ModelManager
      */
-    protected $manager = null;
+    protected $manager;

     /**
      * {@inheritdoc}
@@ -76,17 +76,6 @@
         ];
     }

-    /**
-     * Enable json renderer for index / load action
-     * Check acl rules
-     */
-    public function preDispatch()
-    {
-        if ($this->Request()->getActionName() !== 'upload') {
-            parent::preDispatch();
-        }
-    }
-
     /**
      * Returns a JSON string containing all media albums.
      * Unlike the other Shopware backend controller actions, this action uses the standard method "find".
@@ -411,10 +400,15 @@
             /** @var $file UploadedFile */
             $file = $fileBag->get('fileId');
         } catch (Exception $e) {
-            die(json_encode(['success' => false, 'message' => $e->getMessage()]));
+            $this->View()->assign(['success' => false, 'message' => $e->getMessage()]);
+
+            return;
         }
+
         if ($file === null) {
-            die(json_encode(['success' => false]));
+            $this->View()->assign(['success' => false]);
+
+            return;
         }

         // Create a new model and set the properties
@@ -422,7 +416,7 @@

         $albumId = !empty($params['albumID']) ? $params['albumID'] : -10;
         /* @var $album \Shopware\Models\Media\Album */
-        $album = Shopware()->Models()->find(\Shopware\Models\Media\Album::class, $albumId);
+        $album = $this->getModelManager()->find(\Shopware\Models\Media\Album::class, $albumId);

         if (!$album) {
             $this->View()->assign(['success' => false, 'message' => 'Invalid album id passed']);
@@ -434,38 +428,35 @@
         $media->setDescription('');
         $media->setCreated(new DateTime());

-        $identity = Shopware()->Container()->get('Auth')->getIdentity();
+        $identity = $this->get('Auth')->getIdentity();
         if ($identity !== null) {
             $media->setUserId($identity->id);
         } else {
             $media->setUserId(0);
         }

-        $this->Response()->setHeader('Content-Type', 'text/plain');
-
         try {
             // Set the upload file into the model. The model saves the file to the directory
             $media->setFile($file);

             // Persist the model into the model manager
-            Shopware()->Models()->persist($media);
-            Shopware()->Models()->flush();
+            $this->getModelManager()->persist($media);
+            $this->getModelManager()->flush();
             $data = $this->getMedia($media->getId())->getQuery()->getArrayResult();

             if ($media->getType() === Media::TYPE_IMAGE && // GD doesn't support the following image formats
                 !in_array($media->getExtension(), ['tif', 'tiff'], true)) {
-                $manager = Shopware()->Container()->get('thumbnail_manager');
+                $manager = $this->get('thumbnail_manager');
                 $manager->createMediaThumbnail($media, [], true);
             }

-            $mediaService = Shopware()->Container()->get('shopware_media.media_service');
-            $data[0]['path'] = $mediaService->getUrl($data[0]['path']);
+            $data[0]['path'] = $this->get('shopware_media.media_service')->getUrl($data[0]['path']);

-            die(json_encode(['success' => true, 'data' => $data[0]]));
+            $this->View()->assign(['success' => true, 'data' => $data[0]]);
         } catch (\Exception $e) {
             unlink($file->getPathname());

-            die(json_encode(['success' => false, 'message' => $e->getMessage(), 'exception' => $this->parseExceptionForResponse($e)]));
+            $this->View()->assign(['success' => false, 'message' => $e->getMessage(), 'exception' => $this->parseExceptionForResponse($e)]);
         }
     }

Index: engine/Shopware/Controllers/Backend/Order.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- engine/Shopware/Controllers/Backend/Order.php	(date 1540825291000)
+++ engine/Shopware/Controllers/Backend/Order.php	(date 1541163784000)
@@ -1796,7 +1796,7 @@

         if ($renderer === 'html') {
             exit;
-        } // Debu//g-Mode
+        }

         return true;
     }
Index: engine/Shopware/Controllers/Frontend/Account.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- engine/Shopware/Controllers/Frontend/Account.php	(date 1540825291000)
+++ engine/Shopware/Controllers/Frontend/Account.php	(date 1541163784000)
@@ -449,7 +449,9 @@
         $upstream = $filesystem->readStream($filePath);
         $downstream = fopen('php://output', 'wb');

-        ob_end_clean();
+        if (!$this->Front()->getParam('disableOutputBuffering')) {
+            ob_end_clean();
+        }

         while (!feof($upstream)) {
             fwrite($downstream, fread($upstream, 4096));
